course_catalog/views.py

```diff
@@ -328,3 +339,40 @@ class TopicViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = CourseTopicSerializer
     pagination_class = LargePagination
     permission_classes = (AnonymousAccessReadonlyPermission,)
+
+
+class WebhookOCWView(APIView):
+    """
+    Handle webhooks coming from the OCW bucket
+    """
+
+    permission_classes = ()
+    authentication_classes = ()
```

Comment:
Is there any way to authenticate that the webhook came from S3?

Reply:
@markbreedlove is there some kind of extra authentication key parameter that could always be sent with an SNS notification?  The code could check that the origin is amazon.com but that can be spoofed.  For now I'll add a check that the bucket in the request is the same as `OCW_CONTENT_BUCKET_NAME`

Reply:
It's not clear from the documentation, but I will assume that I can add a querystring to the URL of the request that SNS will send to your endpoint.

Reply:
@markbreedlove can you give that a try, maybe something like `?webhook_key=<somerandomstring>`?  If it works I'll add a new .env `WEBHOOK_KEY` set to that string

Reply:
OK. I've just added an SNS subscription that posts to `discussions-ci`. I can discuss the details with you over Slack. The subscription is currently "pending confirmation."

Reply:
@markbreedlove is there a way to retrieve the confirmation URL?  Otherwise we can try sending another confirmation message once it's on CI, and it should log an exception containing the message body in sentry.

Reply:
We'll have to do the latter, because I can't retrieve the confirmation URL.
I should have looked at the documentation earlier.
I'll need to take the information that's logged and paste it into a form in the control panel to confirm the subscription.

An additional feature, which I'm not inclined to request right now, would be for `WebhookOCWView.post()` to log a specific message or trigger an alert of some kind if the `x-amz-sns-message-type` header is "SubscriptionConfirmation"